server/src/main/java/org/opensearch/common/cache/stats/DefaultCacheStatsHolder.java

```diff
@@ -136,12 +136,12 @@ private boolean internalIncrementHelper(
         List<String> dimensionValues,
         Node node,
         int depth, // Pass in the depth to avoid having to slice the list for each node.
-        Consumer<Node> adder,
+        BiConsumer<Node, Integer> adder,
         boolean createNodesIfAbsent
     ) {
         if (depth == dimensionValues.size()) {
             // This is the leaf node we are trying to reach
-            adder.accept(node);
+            adder.accept(node, depth);
             return true;
```

Comment:
Ideally, having reference to node should be sufficient to validate if it is leaf node or not. Some of the options to verify:

* Storing boolean within each node
* Storing depth within each node
* Validating if the node points to `EMPTY_CHILDREN_MAP`